store/sqlstore/supplier.go

```diff
@@ -176,9 +178,12 @@ func NewSqlSupplier(settings model.SqlSettings, metrics einterfaces.MetricsInter
 	supplier.stores.productNotices = newSqlProductNoticesStore(supplier)
 	err := supplier.GetMaster().CreateTablesIfNotExists()
 	if err != nil {
-		mlog.Critical("Error creating database tables.", mlog.Err(err))
-		time.Sleep(time.Second)
-		os.Exit(EXIT_CREATE_TABLE)
+		if dupError, ok := errors.Cause(err).(*pq.Error); ok && dupError.Code.Name() == DUP_ENTRY_ERROR {
+			mlog.Warn("Duplicate key error occurred.", mlog.Err(dupError))
```

Comment:
nit: perhaps something like this would be a little clearer to anyone seeing this log message

```suggestion
			mlog.Warn("Duplicate key error occurred; assuming table already created and proceeding.", mlog.Err(dupError))
```

Not a required change. Just thinking it may give some clarity.